sound/soc/sof/intel/hda-dsp.c

```diff
@@ -239,3 +239,53 @@ int hda_dsp_core_reset_power_down(struct snd_sof_dev *sdev,
 	return ret;
 }
 
+int hda_dsp_suspend(struct snd_sof_dev *sdev, int state)
+{
+	struct sof_ipc_pm_ctx pm_ctx;
+	int ret = 0;
+
+	memset(&pm_ctx, 0, sizeof(pm_ctx));
+
+	/* configure PM ipc message */
+	pm_ctx.hdr.size = sizeof(pm_ctx);
+	pm_ctx.hdr.cmd = SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_CTX_SAVE;
+
+	/* TODO: uncomment send ipc once FW D0->D3 is complete */
+
+#if 0
+	/* send ipc message */
```

Comment:
Best to try and send this today as it allows DSP to enter D3 ready state. Any context can be saved later.